expect/mod.ts

```diff
@@ -85,3 +85,4 @@
  */
 export { expect } from "./expect.ts";
 export { fn } from "./fn.ts";
+export { iterableEquality } from "./iterable_equality.ts";
```

Comment:
I don't think it's good idea to expose iterableEquality from here.

`iterableEquality` is exported from `@jest/expect-utils`, but the README of `@jest/expect-utils` seems discouraging importing things directly from it. ref. https://www.npmjs.com/package/@jest/expect-utils

I'll make `iterableEquality` private API in `std/expect`

Reply:
I agree in some degree, because it doesn't indeed belong to `expect` module. But if it's private, how to use in public?

Reply:
I suggest we shouldn't provide direct import of `iterableEquality`. The purpose of `std/expect` is compatiblity to jest's expect, but jest doesn't document `iterableEquality` in their docs. Supporting `iterableEquality` is out of scope here in my view.